src/torchjd/backward/multi_task_backward.py

```diff
@@ -0,0 +1,131 @@
+        ``requires_grad`` flag set to ``True``.
+    :param shared_activations: The last shared activation of all tasks. Should be non-empty.
+    :param tasks_parameters: The tensors with respect to which the gradients must be computed. There
+        must be one collection of parameters per task, therefore the length must match with
+        ``tasks_losses``. These must have their ``requires_grad`` flag set to ``True``.
+    :param A: Aggregator to use for the aggregation of the Jacobian.
+    :param retain_graph: If ``False``, the graph used to compute the grad will be freed. Defaults to
+        ``False``.
+    :param parallel_chunk_size: The number of scalars to differentiate simultaneously in the
+        backward pass. If set to ``None``, all coordinates of ``tensors`` will be differentiated in
+        parallel at once. If set to `1`, all coordinates will be differentiated sequentially. A
+        larger value results in faster differentiation, but also higher memory usage. Defaults to
+        ``None``.
+    """
+    _check_optional_positive_chunk_size(parallel_chunk_size)
+
+    shared_activations = _as_tensor_list(shared_activations)
+
+    if len(shared_activations) == 0:
+        raise ValueError("`activations` cannot be an empty iterable of `Tensor`s.")
+
+    _check_retain_graph_compatible_with_chunk_size(
+        shared_activations, retain_graph, parallel_chunk_size
+    )
+
+    if len(tasks_losses) == 0:
+        raise ValueError("`tasks_losses` cannot be an empty sequence of `Tensors`s.")
+    if len(tasks_losses) != len(tasks_parameters):
+        raise ValueError("`tasks_losses` and `tasks_parameters` should have the same size.")
+
+    shared_parameters = list(shared_parameters)
+    tasks_parameters = [list(task_input) for task_input in tasks_parameters]
+
+    # Transforms that store gradient of the losses w.r.t. tasks specific parameters into their
+    # ``.grad`` fields and backpropagate the gradient of the losses w.r.t. to the shared activation.
+    task_transforms = [
+        _make_task_transform(
+            shared_activations,
+            task_parameters,
+            tensor,
+            retain_graph,
+        )
+        for task_parameters, tensor in zip(tasks_parameters, tasks_losses)
+    ]
+
+    # Transform that stacks the gradients of the losses w.r.t. activations into a Jacobian
+    stack = Stack(task_transforms)
+
+    # Transform that computes the Jacobians of the shared parameters
+    jac = Jac(shared_activations, shared_parameters, parallel_chunk_size, retain_graph)
+
+    # Transform that defines the aggregation of the jacobians into gradients
+    aggregation = make_aggregation(UnifyingStrategy(A, shared_parameters))
+
+    # Transform that stores the gradients with respect to the shared parameters
+    store = Store(shared_parameters)
+
+    backward_transform = store << aggregation << jac << stack
```

Comment:
Really nice, maybe we could rename a few things but I like the implementation

Reply:
should we rename `aggregation` to `aggregate` ?